flax/optim/rmsprop.py

```diff
@@ -25,42 +25,55 @@ class _RMSPropHyperParams:
   learning_rate: float
   beta2: float
   eps: float
+  centered: bool
 
 
 @struct.dataclass
 class _RMSPropParamState:
   """RMSProp parameter state"""
 
   v: onp.ndarray
+  mg: onp.ndarray
 
 
 class RMSProp(OptimizerDef):
   """RMSProp optimizer"""
-  def __init__(self, learning_rate: float = None, beta2=0.9, eps=1e-8):
+  def __init__(self, learning_rate: float = None, beta2=0.9, eps=1e-8,
+               centered=False):
     """Constructor for the RMSProp optimizer
-    
+
     Args:
       learning_rate: the step size used to update the parameters.
       beta2: the coefficient used for the moving average of the
         gradient magnitude (default: 0.9).
       eps: the term added to the gradient magnitude estimate for
         numerical stability.
+      centered: If `True`, gradients are normalized by the estimated
+        variance of the gradient; if False, by the uncentered second moment.
+        Setting this to `True` may help with training, but is slightly more
+        expensive in terms of computation and memory. Defaults to `False`.
     """
-    hyper_params = _RMSPropHyperParams(learning_rate, beta2, eps)
+    hyper_params = _RMSPropHyperParams(learning_rate, beta2, eps, centered)
     super().__init__(hyper_params)
 
   def init_param_state(self, param):
     """Initialize parameter state"""
-    return _RMSPropParamState(jnp.zeros_like(param))
+    return _RMSPropParamState(jnp.zeros_like(param), jnp.zeros_like(param))
```

Comment:
This seems wasteful to me. If centered is `False` I would assume we simple set `mg = None`

Reply:
Done.
I've also added tests for the centered version.
TALP?